### PR TITLE
PSE ref don't change if we change product ref

### PR DESCRIPTION
**Pull request by InformatiqueProg**

### CODE REVIEW

File: core/lib/Thelia/Action/Product.php [hunk 1]
```diff
@@ -110,7 +112,14 @@ public function update(ProductUpdateEvent $event)
                     ->save($con)
                 ;
 
-                // Update default category (ifd required)
+                // Update default PSE (if product has no attributes and the product's ref change)
+                if ($prev_ref != $product->getRef() && $product->countSaleElements() == 1) {
```
Comment (Mertiozys) [inline core/lib/Thelia/Action/Product.php]: This checks if there is 1 PSE, not if there are no attributes
What if there is 1 attribute combination ? :-)
You should add a condition to your test if $product->getDefaultSaleElements()->countAttributeCombinations() == 0

File: core/lib/Thelia/Action/Product.php [hunk 2]
```diff
@@ -95,6 +95,8 @@ public function update(ProductUpdateEvent $event)
             $con->beginTransaction();
 
             try {
+                $prev_ref = $product->getRef();
```
Comment (Mertiozys) [inline core/lib/Thelia/Action/Product.php]: Please use camelCase instead of underscore_case

File: core/lib/Thelia/Action/Product.php (hunk 1 above)
Comment (InformatiqueProg) [inline core/lib/Thelia/Action/Product.php]: Create a product with 1 attribute combination?  :'-(
After all why not

Comment (Mertiozys) [inline core/lib/Thelia/Action/Product.php]: Yes that's weird but this case has to be taken into account :-)

**Comment by bibich** [comment]
Hi.
#1790 fixed this issue. 
So I close your PR.
Thanks.

**Pull request closed without merging**